File: Cafeinated.Backend/Cafeinated.Backend.Core/Entities/CoffeeShop.cs
using System.ComponentModel.DataAnnotations.Schema;
using Cafeinated.Backend.Core.Entities.Abstractions;

namespace Cafeinated.Backend.Core.Entities;

public class CoffeeShop : BaseEntity
{
    public string Name { get; set; }
    public string Address { get; set; }
    public string Website { get; set; }
    
    [ForeignKey("PhotoPreviewId")]
    public File PhotoPreview { get; set; }
    public string PhotoPreviewId { get; set; }
    
    public float AverageRating { get; set; }
}

File: Cafeinated.Frontend/src/app/common/coffee-shop.ts
import {Coffee} from "./coffee";

export class CoffeeShop {
  constructor(
    public id: string,
    public name: string,
    public photoPreviewUrl: string ,
    public address: string,
    public website: string,
    public mapImageUrl: string ,
    public coffeeList:Coffee[],
    public averageRating: number) {
  }
}
